chapters/sparse.rst

```diff
@@ -0,0 +1,272 @@
+In contains an algorithm that can be used to perform fast approximate search with sparse inputs.
+Developed by Facebook AI Research.
+
+Build your Sparse Pipeline In Action
+--------------------------------------
+
+In this pipeline, we will make use of Jina's ``TFIDFTextEncoder`` together with ``PysparnnIndexer`` for encoding and indexing.
+
+Step 1. Vectorize your data into sparse vector encoding
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As was mentioned before, ``TFIDFTextEncoder`` was created based on Scikit-learn,
+before using the Encoder, you need to ``fit`` the vectorizer with your training data.
+In this example, we use a simple corpus containing four sentences of text.
+
+.. highlight:: python
+.. code-block:: python
+
+    import pickle
+    from sklearn.feature_extraction.text import TfidfVectorizer
+
+    corpus = [
+        'This is the first document.',
+        'This document is the second document.',
+        'And this is the third one.',
+        'Is this the first document?'
+    ]
+
+    vectorizer = TfidfVectorizer()
+    vectorizer.fit(corpus)
+    # Dump the vectorizer fitted on your training data.
+    pickle.dump(tfidf_vectorizer, open("./tfidf_vectorizer.pickle", "wb"))
+
+Step 2. Setup Encoder & Indexer YAML configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. highlight:: yaml
+.. code-block:: yaml
+
+    !TFIDFTextEncoder
+    metas:
+      name: tfidf_encoder
+    with:
+      path_vectorizer: ./tfidf_vectorizer.pickle
+
+For the indexer,
+we will use the ``PysparnnIndexer`` with approximate nearest neighbor for sparse data.
+Since we want to store the indexed result, we combined ``PysparnnIndexer`` and ``BinaryPbIndexer`` together.
+
+.. highlight:: yaml
+.. code-block:: yaml
+
+    !CompoundIndexer
+    components:
+      - !PysparnnIndexer
+        with:
+          prefix_filename: 'pysparnn'
+        metas:
+          name: vecidx
+      - !BinaryPbIndexer
+        with:
+          index_filename: doc.gz
+        metas:
+          name: docidx
+    metas:
+      name: doc_compound_indexer
+      workspace: $WORKDIR
+
+Step 3. Create your index flow
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. highlight:: yaml
+.. code-block:: yaml
+
+    jtype: Flow
+    pods:
+      encoder:
+        uses: encode.yml
+        show_exc_info: true
+        parallel: 1
+        timeout_ready: 600000
+        read_only: true
+      doc_indexer:
+        uses: indexer.yml
+        shards: 1
+        separated_workspace: true
+
+Step 4. Create your query flow
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. highlight:: yaml
+.. code-block:: yaml
+
+    jtype: Flow
+    with:
+      read_only: true
+    pods:
+      encoder:
+        uses: encode.yml
+        parallel: 1
+        timeout_ready: 600000
+        read_only: true
+      doc_indexer:
+        uses: indexer.yml
+        shards: 1
+        separated_workspace: true
+        polling: all
+        uses_reducing: _merge_all
```

Comment:
this keyword is all, as well as `separated_workspace`. Since `shards` = 1 no need to add `uses_after` or `polling` or anything